includes/Modules/Analytics.php

```diff
@@ -1331,4 +1365,97 @@ public function check_service_entity_access() {
 		return true;
 	}
 
+	/**
+	 * Validates the report metrics.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param array $metrics The metrics to validate.
+	 *
+	 * @return null|WP_Error NULL if metrics are valid, otherwise WP_Error.
+	 */
+	protected function validate_report_metrics( $metrics ) {
+		if ( false === $this->is_using_shared_credentials ) {
+			return null;
+		}
+
+		$valid_metrics = apply_filters(
+			'googlesitekit_shareable_analytics_metrics',
+			array(
+				'ga:sessions',
+				'ga:users',
+				'ga:pageviews',
+				'ga:uniquePageviews',
+				'ga:bounceRate',
+				'ga:avgSessionDuration',
+				'ga:adsenseRevenue',
+				'ga:adsenseECPM',
+				'ga:adsensePageImpressions',
+				'ga:goalCompletionsAll',
+			)
+		);
+
+		$invalid_metrics = array_diff( $metrics, $valid_metrics );
+
+		if ( count( $invalid_metrics ) > 0 ) {
+			return sprintf(
+				/* translators: %s is replaced with a comma separated list of the invalid metrics. */
+				_n(
+					'Unsupported metric requested: %s',
+					'Unsupported metrics requested: %s',
+					count( $invalid_metrics ),
+					'google-site-kit'
+				),
+				implode( ', ', $invalid_metrics )
```

Comment:
Ah, thanks. My understanding is that won't always map well to all locales, but I suppose it's better than nothing, and since we have it already worth using. 👍🏻

Reply:
Addressed this in #5727, thank you!